### DISCUSSION TITLE
xtracfg: How do I use this tool in the windows environment?

### DISCUSSION THREAD

**Discussion by Sverre-Iversen** [ldproxy/ldproxy#1172] [Configuration]
**I have tried to set up xtracfg with help of https://docs.ldproxy.net/cli/xtracfg/, but with no luck. Docker is quite new to me.**

**1**) My **ldproxy** is set up like this - **it works fine**:
**docker run --name ldproxy -d -p 7080:7080 -v C:/docker/ldproxy_data:/ldproxy/data iide/ldproxy:next**

**2**) Installation of **xtracfg**:
**docker pull ghcr.io/ldproxy/xtracfg**
```
Using default tag: latest
latest: Pulling from ldproxy/xtracfg
6c1f3ba730bd: Pull complete
2e1227421b71: Pull complete
b6c5cb566fc8: Pull complete
Digest: sha256:7d79db21336cdc32a3a6024120b81d133a23d44a47b44a0745e2685fd9ed1ff2
Status: Downloaded newer image for ghcr.io/ldproxy/xtracfg:latest
ghcr.io/ldproxy/xtracfg:latest

What's Next?
  View a summary of image vulnerabilities and recommendations → docker scout quickview ghcr.io/ldproxy/xtracfg
```

**3**) Now I try to run **xtracfg**:
**docker run -it --rm -v C:/docker/ldproxy_data:/ldproxy/data ghcr.io/ldproxy/xtracfg info**
`<no text are shown>`

**I am clearly missing something - is the volume mapping wrong or/and a missing port setting?**

**Comment by azahnen** [comment]
The mount directory inside of the container is different than for ldproxy, it has to be `/src`. So the full command should look like this:

```
docker run -it --rm -v C:/docker/ldproxy_data:/src ghcr.io/ldproxy/xtracfg info
```

**Reply by Sverre-Iversen** [reply]
I tried your suggestion, but with no luck. After the "docker pull ghcr.io/ldproxy/xtracfg" command I tried this:
`docker run -it --rm ghcr.io/ldproxy/xtracfg --help`
... and nothing happens

I can see that the container is shown for a second in "Docker desktop" before it terminates (disappears).
Seems to be lacking some extra instructions for xtracfg in the Windows environment.
Any ideas?

**Reply by azahnen** [reply]
Yeah, that looks like something is wrong with the image. I will look into this.

**Reply by Sverre-Iversen** [reply]
**Temporarily solution - I learned how to it manually in a shell inside the container:**
docker exec -it ldproxy  /bin/bash
root@99207b22ef8f:/ldproxy# cd data
#Go to download directory
root@99207b22ef8f:/ldproxy/data# cd /tmp
root@99207b22ef8f:/tmp# wget https://github.com/interactive-instruments/xtraplatform-cli/releases/download/v2.0.1/xtracfg_v2.0.1_linux_amd64.tar.gz
root@99207b22ef8f:/tmp#tar xvzf xtracfg_v2.*.tar.gz 

#the commands are now possible to run:
root@99207b22ef8f:/ldproxy/data# /tmp/xtracfg --src /ldproxy/data check --help
